src/core/radio/ble_secure.hpp

```diff
@@ -0,0 +1,499 @@
+     *
+     * @retval kErrorNone          Successfully sent data.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error SendApplicationTlv(uint8_t *aBuf, uint16_t aLength);
+
+    /**
+     * This method sends all remaining bytes in the send buffer.
+     *
+     * @retval kErrorNone          Successfully enqueued data into the output interface.
+     * @retval kErrorNoBufs        Failed to allocate buffer memory.
+     * @retval kErrorInvalidState  TLS connection was not initialized.
+     *
+     */
+    Error Flush();
+
+    /**
+     * This method is used to pass data received over a BLE link to the secure BLE server.
+     *
+     * @param[in]  aBuf            A pointer to the data received.
+     * @param[in]  aLength         A number indicating the length of the data buffer.
+     *
+     */
+    Error HandleBleReceive(uint8_t *aBuf, uint16_t aLength);
+
+    /**
+     * This method is used to notify the secure BLE server that a BLE Device has been connected.
+     *
+     * @param[in]  aConnectionId    The identifier of the open connection.
+     *
+     */
+    Error HandleBleConnected(uint16_t aConnectionId);
+
+    /**
+     * This method is used to notify the secure BLE server that the BLE Device has been disconnected.
+     *
+     * @param[in]  aConnectionId    The identifier of the open connection.
+     *
+     */
+    Error HandleBleDisconnected(uint16_t aConnectionId);
+
+private:
+    static constexpr uint8_t  kInitialMtuSize   = 23; // ATT_MTU
+    static constexpr uint8_t  kGattOverhead     = 3;  // BLE GATT payload fits MTU size - 3 bytes
+    static constexpr uint8_t  kPacketBufferSize = 64; // must be >= kInitialMtuSize - kGattOverhead
+    static constexpr uint16_t kTxBleHandle      = 0;  // Characteristics Handle for TX (not used)
+    static constexpr uint16_t kFlushThreshold =
+        256; // mSentMessage length exceeding this threshold triggers flushing the buffer
+
+    static void HandleTlsConnected(void *aContext, bool aConnected);
+    void        HandleTlsConnected(bool aConnected);
+
+    static void HandleTlsReceive(void *aContext, uint8_t *aBuf, uint16_t aLength);
+    void        HandleTlsReceive(uint8_t *aBuf, uint16_t aLength);
+
+    static void HandleTransmit(Tasklet &aTasklet);
+    void        HandleTransmit(void);
+
+    static Error HandleTransport(void *aContext, ot::Message &aMessage, const Ip6::MessageInfo &aMessageInfo);
+    Error        HandleTransport(ot::Message &aMessage);
+
+    MeshCoP::Dtls             mTls;
+    MeshCoP::TcatAgent        mTcatAgent;
+    Callback<ConnectCallback> mConnectCallback;
+    Callback<ReceiveCallback> mReceiveCallback;
+    bool                      mTlvMode;
+    ot::Message              *mReceivedMessage;
+    ot::Message              *mSentMessage;
+    ot::MessageQueue          mTransmitQueue;
+    TaskletContext            mTransmitTask;
```

Comment:
Can use `TaskletIn<>` instead of `TaskletContext`?

```c++
   using TxTask = TaskletIn<BleSecure, &BleSecure::HandleTransmit>;

   ...
   TxTask  mTransmitTask;
```
And we don't need the `static void HandleTransmit(Tasklet &aTasklet);` anymore.

Reply:
done